src/gen_rules.ml

```diff
@@ -211,12 +211,20 @@ module Gen(P : Install_rules.Params) = struct
     executables_rules t.exes ~dir ~scope ~dir_kind
       ~dir_contents
 
+  let gen_format_rules sctx ~dir ~scope =
+    let project = Scope.project scope in
+    match Dune_project.find_extension_args project Auto_format.key with
+    | None -> ()
+    | Some config ->
+      Format_rules.gen_rules sctx config ~dir ~scope
+
   (* +-----------------------------------------------------------------+
      | Stanza                                                          |
      +-----------------------------------------------------------------+ *)
 
   let gen_rules dir_contents
         { SC.Dir_with_jbuild. src_dir; ctx_dir; stanzas; scope; kind } =
+    gen_format_rules sctx ~dir:ctx_dir ~scope;
```

Comment:
What would be a good place to call this setup code? It's a bit difficult to understand what the different parts are doing.

Reply:
In the last `gen_rules` function in this module, in the `match Dir_contents.kind dir_contents with`, we iterate over all directories of the group at this point.

The code is a bit complex indeed, it could use some refactoring

Reply:
I tried to call it just before that match. Is this the way to go? Also there's no `scope`, so I need to rebuild it from the `sctx` here.

Reply:
We also need to call it for every sub-directory in the case `Group_root`.  Looking at this again, we could also call it in the `gen_rules` function that comes just before since this one is called once for every directory in the group.